src/ir_Corona.cpp

```diff
@@ -0,0 +1,578 @@
+uint16_t IRCoronaAc::getOnTimer(void) {
+  return _getTimer(kCoronaAcOnTimerSection);
+}
+
+/// Set the On Timer time
+/// @param[in] nr_of_mins Number of minutes to set the timer to.
+///   (0 or kCoronaAcTimerOff is disable).
+void IRCoronaAc::setOnTimer(const uint16_t nr_of_mins) {
+  _setTimer(kCoronaAcOnTimerSection, nr_of_mins);
+  // if we set a timer value, clear the other timer
+  if (getOnTimer())
+    setOffTimer(kCoronaAcTimerOff);
+}
+
+/// Get the current Off Timer time
+/// @return The number of minutes it is set for. 0 means it's off.
+uint16_t IRCoronaAc::getOffTimer(void) {
+  return _getTimer(kCoronaAcOffTimerSection);
+}
+
+/// Set the Off Timer time
+/// @param[in] nr_of_mins Number of minutes to set the timer to.
+///   (0 or kCoronaAcTimerOff is disable).
+void IRCoronaAc::setOffTimer(const uint16_t nr_of_mins) {
+  _setTimer(kCoronaAcOffTimerSection, nr_of_mins);
+  // if we set a timer value, clear the other timer
+  if (getOffTimer())
+    setOnTimer(kCoronaAcTimerOff);
+}
+
+/// Convert the internal state into a human readable string.
+/// @return The current internal state expressed as a human readable String.
+String IRCoronaAc::toString(void) {
+  String result = "";
+  result.reserve(120);  // Reserve some heap for the string to reduce fragging.
+  result += addBoolToString(getPower(), kPowerStr, false);
+  result += addBoolToString(getPowerToggle(), kPowerToggleStr);
+  result += addModeToString(getMode(), 0xFF, kCoronaAcModeCool,
+                            kCoronaAcModeHeat, kCoronaAcModeDry,
+                            kCoronaAcModeFan);
+  result += addTempToString(getTemp());
+  result += addFanToString(getFan(), kCoronaAcFanHigh, kCoronaAcFanLow,
+                           kCoronaAcFanAuto, kCoronaAcFanAuto,
+                           kCoronaAcFanMedium);
+  result += addBoolToString(getSwingV(), kSwingVToggleStr);
+  result += addBoolToString(getEcono(), kEconoStr);
+  result += addLabeledString(getOnTimer()
+                             ? minsToString(getOnTimer()) : kOffStr,
+                             kOnTimerStr);
+  result += addLabeledString(getOffTimer()
+                             ? minsToString(getOffTimer()) : kOffStr,
+                             kOffTimerStr);
+  return result;
+}
+
+/// Convert the A/C state to it's common stdAc::state_t equivalent.
+/// @return A stdAc::state_t state.
+stdAc::state_t IRCoronaAc::toCommon(const stdAc::state_t *prev) {
+  stdAc::state_t result;
+  if (prev != NULL) {
+    result = *prev;
+  } else {
+    result.swingv = stdAc::swingv_t::kOff;
+  }
+  result.protocol = decode_type_t::CORONA_AC;
+  result.model = -1;  // No models used.
+  result.power = getPower();
+  result.mode = toCommonMode(getMode());
+  result.celsius = true;
+  result.degrees = getTemp();
+  result.fanspeed = toCommonFanSpeed(getFan());
+  if (getSwingV()) {
+    result.swingv = result.swingv != stdAc::swingv_t::kOff ?
```

Comment:
ProTip:
`   result.swingv = (result.swingv != stdAc::swingv_t::kOff) ?`
I've often seen the conditional assignment chose poorly what to condition off.
e.g.
is it:
`   result.swingv = result.swingv != (stdAc::swingv_t::kOff ? stdAc::swingv_t::kOff : stdAc::swingv_t::kAuto);` or
`   result.swingv = (result.swingv != stdAc::swingv_t::kOff) ? stdAc::swingv_t::kOff : stdAc::swingv_t::kAuto;`.

I've spent many a hand full of hair trying to work out what was wrong with things like that. It *looked* right to me, but the compiler saw differently. Best to make it obvious.

Reply:
When it is only one value I have never had an issue, but with more values certainly yes, bitten me to, will change.
(This was a copy from Coolix, search seeing the same in Midea as well.)

Reply:
If it works, and you're sure, feel free to leave it as is.